src/compose/table.c

```diff
@@ -183,6 +183,15 @@ xkb_compose_table_new_from_locale(struct xkb_context *ctx,
             goto found_path;
     }
 
+    cpath = path = get_xdg_xcompose_file_path();
```

Comment:
unrelated to this patch but still nice if you could fix this: the `cpath` vs `path` confusion can be fixed by a single `strdup_safe()` call in `get_xcomposefile_path()`

Reply:
Done. :)